lib/msal-core/src/ScopeSet.ts

```diff
@@ -16,9 +16,10 @@ export class ScopeSet {
      */
     // TODO: Rename this, intersecting scopes isn't a great name for duplicate checker
     static isIntersectingScopes(cachedScopes: Array<string>, scopes: Array<string>): boolean {
-        cachedScopes = this.convertToLowerCase(cachedScopes);
-        for (let i = 0; i < scopes.length; i++) {
-            if (cachedScopes.indexOf(scopes[i].toLowerCase()) > -1) {
+        const convertedCachedScopes = this.trimAndConvertToLowerCase([...cachedScopes]);
+        const requestScopes = this.trimAndConvertToLowerCase([...scopes]);
+        for (let i = 0; i < requestScopes.length; i++) {
```

Comment:
not a huge deal here, but calling out this is n^2

Reply:
Making this linear complexity is going to require implementing a hash table, because the scope arrays are unsorted and there is no way to order scopes.

Reply:
Yeah I think its fine if its exponential, there are limited scopes so its not a huge deal